File: lib_base/src/main/java/com/czl/lib_base/annotation/TodoOrder.java
package com.czl.lib_base.annotation;

import androidx.annotation.IntDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * @author Alwyn
 * @Date 2021/1/6
 * @Description
 */
@IntDef({TodoOrder.finishAsc,TodoOrder.finishDesc,TodoOrder.createAsc,TodoOrder.createDesc})
@Retention(RetentionPolicy.SOURCE)
public @interface TodoOrder {
    // 完成日期顺序
    int finishAsc = 1;
    // 倒序
    int finishDesc = 2;
    // 创建日期顺序
    int createAsc = 3;
    int createDesc = 4;
}

File: lib_base/src/main/java/com/czl/lib_base/annotation/TodoOrder.kt
package com.czl.lib_base.annotation

import androidx.annotation.IntDef

/**
 * @author Alwyn
 * @Date 2021/1/6
 * @Description
 */
@IntDef(TodoOrder.finishAsc, TodoOrder.finishDesc, TodoOrder.createAsc, TodoOrder.createDesc)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
annotation class TodoOrder {
    companion object {
        // 完成日期顺序
        const val finishAsc = 1
        // 倒序
        const val finishDesc = 2
        // 创建日期顺序
        const val createAsc = 3
        const val createDesc = 4
    }
}